datatransfer/impl/graphsync/graphsync.go

```diff
@@ -194,15 +202,20 @@ func (receiver *graphsyncReceiver) ReceiveRequest(
 	sender peer.ID,
 	incoming message.DataTransferRequest) {
 
-	var success bool
 	// not yet doing anything else with the voucher
 	_, err := receiver.validateVoucher(sender, incoming)
-	if err == nil {
-		success = true
+	if err != nil {
+		receiver.impl.sendResponse(ctx, false, sender, incoming.TransferID())
+		return
 	}
-
-	// not yet doing anything else with the transfer ID
-	_, err = receiver.impl.sendResponse(success, sender, incoming.TransferID())
+	stor, _ := nodeFromBytes(incoming.Selector())
+	root := cidlink.Link{incoming.BaseCid()}
+	ctx, cancel := context.WithCancel(ctx)
+	go func() {
```

Comment:
graphsync.request is non blocking cause it returns a channel. it doesn't need to be in a go routine.

Reply:
also, this can be addressed in the next ticket, but you don't want to schedule a graphsync request here UNLESS its a push. In the case of a pull, the request is scheduled when the response is received.

Reply:
re: scheduling gs req, that was a guess before I saw the next story, but I can fix it here